data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1319,14 +1319,513 @@ Marksman attacks are only affected if the chance to hit is greater than 60%."
+    [event]
+        name = attack
+        first_time_only = no
+
+        [filter_attack]
+            special_id = taunt
+        [/filter_attack]
+
+        [object] 
+            [filter] 
+                find_in = second_unit
+            [/filter]
+
+            [effect]
+                apply_to = new_ability
+
+                [abilities]
+                    {ABILITY_TAUNTED}
+                [/abilities]
+            [/effect]
+        [/object]
+    [/event]
+#enddef
+
+#define ABILITY_TAUNTED
+    [dummy]
+        id = taunted
+        name = _"taunted"
+        description= _ "This unit is taunted by Kaleh, it will try to attack him no matter what."
+    [/dummy]
+#enddef
+
+#define ABILITY_INSPIRE
+    [leadership]
+        id=inspire
+        value="(25 * (level - other.level + 1))"
+        cumulative=no
+        name= _ "inspire"
+        female_name= _ "female^inspire"
+        description= _ "This unit can inspire own units that are next to it, making them fight better. Adjacent own units of lower or equal level will deal 25% more damage plus 25% for each level they are below the inspiring unit."
+        special_note=_"The presence of this unit inspires own units next to it to deal more damage in combat, though this only applies to units of lower or equal level."
+        affect_self=no
+        [affect_adjacent]
+            [filter]
+                formula="level <= other.level"
+            [/filter]
+        [/affect_adjacent]
+    [/leadership]
+#enddef
+
+#define STR_COOLDOWN_NAME_2
+_ "cooldown 1"#enddef
+
+#define STR_COOLDOWN_NAME_3
+_ "cooldown 2"#enddef
+
+#define STR_COOLDOWN_RESTRICT_2
+_ "After using this attack, you cant use it during your next turn"#enddef
+
+#define STR_COOLDOWN_RESTRICT_3
+_ "After using this attack, you cant use it during your next two turn"#enddef
+
+#define WEAPON_SPECIAL_COOLDOWN X
+    [disable]
+        id = cooldown_{X}
+        name = {STR_COOLDOWN_NAME_{X}}
+        description = {STR_COOLDOWN_RESTRICT_{X}}
+        [filter_self]
+            formula = "self.wml_vars.cooldown > 0"
+        [/filter_self]
+        active_on = offense
+    [/disable]
+#enddef
+
+#define WEAPON_SPECIAL_COOLDOWN_EVENTS X
+    [event]
+        name = attack end
+        id = cooldown_{X}
+        first_time_only = no
+
+        [filter_attack]
+            special_id = cooldown_{X}
+        [/filter_attack]
+        [modify_unit]
+            [filter]
+                find_in = unit
+            [/filter]
+
+            [set_variable]
+                name = cooldown
+                value = {X}
+            [/set_variable]
+        [/modify_unit]
+    [/event]
+    [event]
+        name = side 1 turn refresh 
+        id = cooling
+        first_time_only = no
+
+        [store_unit]
+            [filter]
+                [has_attack]
+                    special_id = cooldown_{X}
+                [/has_attack]
+            [/filter]
+
+            variable = cooling_units
```

Comment:
Why does this need to use `[store_unit]` and a loop instead of just putting this filter into `[modify_unit]`?

I'd understand if it was checking for units with `cooldown > 0` and only applying `[modify_unit]` to ones that are cooling down, but it isn't.